pkg/resources/grant_privileges_to_share.go

```diff
@@ -0,0 +1,474 @@
+		ExactlyOneOf:     grantPrivilegesToShareGrantExactlyOneOfValidation,
+	},
+	"view_name": {
+		Type:             schema.TypeString,
+		Optional:         true,
+		ForceNew:         true,
+		Description:      "The fully qualified name of the view on which privileges will be granted.",
+		ValidateDiagFunc: IsValidIdentifier[sdk.SchemaObjectIdentifier](),
+		ExactlyOneOf:     grantPrivilegesToShareGrantExactlyOneOfValidation,
+	},
+}
+
+func GrantPrivilegesToShare() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: CreateGrantPrivilegesToShare,
+		UpdateContext: UpdateGrantPrivilegesToShare,
+		DeleteContext: DeleteGrantPrivilegesToShare,
+		ReadContext:   ReadGrantPrivilegesToShare,
+
+		Schema: grantPrivilegesToShareSchema,
+		Importer: &schema.ResourceImporter{
+			StateContext: ImportGrantPrivilegesToShare(),
+		},
+	}
+}
+
+func ImportGrantPrivilegesToShare() func(ctx context.Context, d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
+	return func(ctx context.Context, d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
+		id, err := ParseGrantPrivilegesToShareId(d.Id())
+		if err != nil {
+			return nil, err
+		}
+		if err := d.Set("share_name", id.ShareName.Name()); err != nil {
+			return nil, err
+		}
+		if err := d.Set("privileges", id.Privileges); err != nil {
+			return nil, err
+		}
+
+		switch id.Kind {
+		case OnDatabaseShareGrantKind:
+			if err := d.Set("database_name", id.Identifier.Name()); err != nil {
+				return nil, err
+			}
+		case OnSchemaShareGrantKind:
+			if err := d.Set("schema_name", id.Identifier.FullyQualifiedName()); err != nil {
+				return nil, err
+			}
+		// case OnFunctionShareGrantKind:
+		//	if err := d.Set("function_name", id.Identifier.FullyQualifiedName()); err != nil {
+		//		return nil, err
+		//	}
+		case OnTableShareGrantKind:
+			if err := d.Set("table_name", id.Identifier.FullyQualifiedName()); err != nil {
+				return nil, err
+			}
+		case OnAllTablesInSchemaShareGrantKind:
+			if err := d.Set("all_tables_in_schema", id.Identifier.FullyQualifiedName()); err != nil {
+				return nil, err
+			}
+		case OnTagShareGrantKind:
+			if err := d.Set("tag_name", id.Identifier.FullyQualifiedName()); err != nil {
+				return nil, err
+			}
+		case OnViewShareGrantKind:
+			if err := d.Set("view_name", id.Identifier.FullyQualifiedName()); err != nil {
+				return nil, err
+			}
+		}
+
+		return []*schema.ResourceData{d}, nil
+	}
+}
+
+func CreateGrantPrivilegesToShare(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	db := meta.(*sql.DB)
+	client := sdk.NewClientFromDB(db)
+	id := createGrantPrivilegesToShareIdFromSchema(d)
+	log.Printf("[DEBUG] created identifier from schema: %s", id.String())
+
+	err := client.Grants.GrantPrivilegeToShare(ctx, getObjectPrivilegesFromSchema(d), getShareGrantOn(d), sdk.NewAccountObjectIdentifier(id.ShareName.Name()))
+	if err != nil {
+		return diag.Diagnostics{
```

Comment:
you can do diag.FromError() instead. Or maybe a helper function would be more appropriate.

Reply:
yeah, but diag.FromError() puts everything in the summary section and I wanted to put specifics in the details. We should probably make some internal helper functions for it to have a function that can create diags with severity, summary and details as arguments.